x/examples/outline-cli/routing_linux.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"net"
+
+	"github.com/vishvananda/netlink"
+)
+
+var ipRule *netlink.Rule = nil
+
+func startRouting(proxyIP string, config *RoutingConfig) error {
+	if err := setupRoutingTable(config.RoutingTableID, config.TunDeviceName, config.TunGatewayCIDR, config.TunDeviceIP); err != nil {
+		return err
+	}
+	return setupIpRule(proxyIP+"/32", config.RoutingTableID, config.RoutingTablePriority)
```

Comment:
You're right.I added a TODO item in `setupIpRule` function below. This might involve more changes because we need to get the underlying socket FD and apply a fwmask to it, this requires some changes to the `transport` package.